src/Analyzers/Orchestration/OrchestrationAnalyzer.cs

```diff
@@ -10,10 +10,95 @@
 
 namespace Microsoft.DurableTask.Analyzers.Orchestration;
 
+/// <summary>
+/// Base class for visitors that analyze orchestrations.
+/// </summary>
+public abstract class OrchestrationVisitor
```

Comment:
For reference, using this abstraction, here's how easy it is to build new analyzers that don't care about method probing, such as the ones only interested in inspecting method's signatures: https://github.com/allantargino/durabletask-dotnet/commit/162a5e04dc73bba678ed11121244279d90383274